src/BuiltInTools/dotnet-watch/WatchCommandLineOptions.cs

```diff
@@ -136,69 +138,74 @@ console output.
         };
 
         var remainingWatchArgs = new CliArgument<string[]>("forwardedArgs") { Description = "Arguments to pass to the child dotnet process." };
-        var remainingRunArgs = new CliArgument<string[]>("remainingRunArgs");
+        var remainingCommandArgs = new CliArgument<string[]>("remainingRunArgs");
 
         var runCommand = new CliCommand("run") { Hidden = true };
+        var testCommand = new CliCommand("test") { Hidden = true };
+        var buildCommand = new CliCommand("build") { Hidden = true };
```

Comment:
this implies that this is the only set of supported commands - why not just have an `string[]` catch-all that is whatever command the user wants to re-trigger on change?

Reply:
Ok, is there a catch-all for all other commands that allows for executing arbitrary commands as well? That's IMO what users expect out of a `watch` command.

Reply:
Another issue is that `dotnet watch x y z` runs the app in Hot Reload mode with arguments `x`, `y` and `z`.
We can't run `build` and `test` commands in Hot Reload mode. So these commands need to be special-cased.

This is an unfortunate design. It'd be better to have a separate verb for Hot Reload instead of using `watch`. 
E.g. `dotnet hotreload`.

Reply:
👍 to `dotnet hotreload` - then wouldn't have had this issue in the first place

Reply:
I think it would also be clearer for the users to have separate commands